x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/retention/ExpiredModelSnapshotsRemoverTests.java

```diff
@@ -97,35 +112,44 @@ public void testRemove_GivenJobsWithMixedRetentionPolicies() {
         Date oneDayAgo = new Date(now.getTime() - TimeValue.timeValueDays(1).getMillis());
         SearchHit snapshot1_1 = createModelSnapshotQueryHit("job-1", "fresh-snapshot", oneDayAgo);
         searchResponses.add(AbstractExpiredJobDataRemoverTests.createSearchResponseFromHits(Collections.singletonList(snapshot1_1)));
+        SearchHit snapshot2_1 = createModelSnapshotQueryHit("job-2", "fresh-snapshot", oneDayAgo);
+        searchResponses.add(AbstractExpiredJobDataRemoverTests.createSearchResponseFromHits(Collections.singletonList(snapshot2_1)));
 
         // It needs to be strictly more than 7 days before the most recent snapshot, hence the extra millisecond
         Date eightDaysAndOneMsAgo = new Date(now.getTime() - TimeValue.timeValueDays(8).getMillis() - 1);
-        SearchHit snapshotToBeDeleted = createModelSnapshotQueryHit("job-1", "old-snapshot", eightDaysAndOneMsAgo);
-
-
-        searchResponses.add(
-            AbstractExpiredJobDataRemoverTests.createSearchResponseFromHits(Collections.singletonList(snapshotToBeDeleted)));
-
-        SearchHit snapshot2_1 = createModelSnapshotQueryHit("job-1", "snapshots-1_1", eightDaysAndOneMsAgo);
-        searchResponses.add(AbstractExpiredJobDataRemoverTests.createSearchResponseFromHits(Collections.singletonList(snapshot2_1)));
-        searchResponses.add(AbstractExpiredJobDataRemoverTests.createSearchResponseFromHits(Collections.emptyList()));
-
-        givenClientRequestsSucceed(searchResponses);
+        Map<String, List<ModelSnapshot>> snapshotResponses = new HashMap<>();
+        snapshotResponses.put("job-1",
+            Arrays.asList(
+                createModelSnapshot("job-1", "active", oneDayAgo),
+                createModelSnapshot("job-1", "fresh-snapshot", oneDayAgo),
```

Comment:
> . But I don't think the code path being tested here is the code path that would get executed in production with the configured settings.

I understand. Including it is just a mocking mistake and the actual code path of "return only things before this timestamp" is not exercised. I could capture the query to check. 

But I think the proof of this thing working is in the true integration tests (I will unmute the one you mentioned)

Reply:
I will update the mocks

But I do think it is nice to test that `active` is not removed (since we are not filtering it in the query anylonger). I will make it `8days` old for `job-1`. An admittedly rare case, but a good one to cover.

Reply:
Yes, it's good to test `active`.  Good idea to make that old enough that it would have been deleted.